infrastructure/code/code_html.go

```diff
@@ -60,32 +62,53 @@ type ExampleCommit struct {
 //go:embed template/details.html
 var detailsHtmlTemplate string
 
-var globalTemplate *template.Template
+type HtmlRenderer struct {
+	c              *config.Config
+	globalTemplate *template.Template
+}
 
-func init() {
+func NewHtmlRenderer(c *config.Config) (*HtmlRenderer, error) {
 	funcMap := template.FuncMap{
 		"repoName":      getRepoName,
 		"trimCWEPrefix": html.TrimCWEPrefix,
 		"idxMinusOne":   html.IdxMinusOne,
 	}
 
-	var err error
-	globalTemplate, err = template.New(string(product.ProductCode)).Funcs(funcMap).Parse(detailsHtmlTemplate)
+	globalTemplate, err := template.New(string(product.ProductCode)).Funcs(funcMap).Parse(detailsHtmlTemplate)
 	if err != nil {
-		config.CurrentConfig().Logger().Error().Msgf("Failed to parse details template: %s", err)
+		c.Logger().Error().Msgf("Failed to parse details template: %s", err)
+		return nil, err
 	}
+
+	return &HtmlRenderer{
+		c:              c,
+		globalTemplate: globalTemplate,
+	}, nil
 }
 
-func getCodeDetailsHtml(issue snyk.Issue, folderPath string) string {
-	c := config.CurrentConfig()
+func determineFolderPath(filePath string) string {
+	ws := workspace.Get()
+	if ws == nil {
+		return ""
+	}
+	for _, folder := range ws.Folders() {
+		folderPath := folder.Path()
+		if strings.HasPrefix(filePath, folderPath) {
+			return folderPath
+		}
+	}
+	return ""
+}
```

Comment:
This implementation is not correct... we got this: `uri.FolderContains` already, so the folder that contains the file, should be the one you want.